### PR TITLE
LP-2429 add mailchimp configuration and api methods

### PR DESCRIPTION
**Pull request by emraan004**

### CODE REVIEW

File: openedx/adg/common/mailchimp_pipeline/apps.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
+"""
+MailchimpPipeline app configurations
```
Comment (muak50) [inline openedx/adg/common/mailchimp_pipeline/apps.py:2]: Space missing between MailChimp and Pipeline

File: openedx/adg/common/mailchimp_pipeline/client.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+        proto = 'https' if secure else 'http'
+        dc = apikey.split('-')[1]
+
+        self.root = '{}://{}.api.mailchimp.com/{}/'.format(proto, dc, self.version)
+
+    def make_request(self, method="GET", path=None, **kwargs):
+        """
+        This function makes the desired request to mailchimp and performs actions on the subscriber list
+        """
```
Comment (muak50) [inline openedx/adg/common/mailchimp_pipeline/client.py:34]: We use Google style docstring. For reference, https://edx.readthedocs.io/projects/edx-developer-guide/en/latest/style_guides/python-guidelines.html

File: openedx/adg/common/mailchimp_pipeline/apps.py [hunk 3]
```diff
@@ -0,0 +1,11 @@
+"""
```
Comment (muak50) [inline openedx/adg/common/mailchimp_pipeline/apps.py:1]: Suggestion: Can we update location of mailchimp_pipeline app to openedx/adg/common**/lib/**mailchimp_pipeline. It seems sort of lib and we have separate directory for that.

**Review by muak50 — APPROVED** [review]
Couple of minor suggestion. Rest LGTM 👍

File: openedx/adg/common/mailchimp_pipeline/apps.py (hunk 1 above)
Comment (ahmed-faraz46) [reply openedx/adg/common/mailchimp_pipeline/apps.py:2]: If this is just a single line we can also write it as
`""" Mailchimp pipeline app configurations """`

File: openedx/adg/common/mailchimp_pipeline/client.py [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+import hashlib
+import json
+
+from django.conf import settings
+from requests import request
+from requests.exceptions import HTTPError
+
+
+class MailChimpException(Exception):
+    pass
```
Comment (ahmed-faraz46) [inline openedx/adg/common/mailchimp_pipeline/client.py:14]: I know we benchmarked the code in PhilU. But I guess we should not override the base `Exception` if are not actually overriding or extending anything in it.

**Review by ahmed-faraz46 — CHANGES_REQUESTED** [review]

**Review by ahmed-faraz46 — APPROVED** [review]
LGTM 👍

File: openedx/adg/common/mailchimp_pipeline/apps.py (hunk 1 above)
Comment (zahid-arbisoft) [reply openedx/adg/common/mailchimp_pipeline/apps.py]: @ahmed-faraz46 This format is against Edx recommendation

**Pull request merged**